the-experts-BE/app.js

```diff
@@ -33,6 +39,47 @@ app.post("/addContractor", function(req, res, next) {
   );
 });
 
+app.post("/addReview", function(req, res, next) {
+  queries.addReview(req.body, (err, dataResponse) => {
+    if (err) next(err);
+    res.json(dataResponse);
+  });
+});
+
+app.post("/login", function(req, res, next) {
+  queries.getUser(req.body.email, (err, dataResponse) => {
+    if (err) next(err);
+    comparePasswords(req.body.password,dataResponse.password,(error,result)=>{
+      if(error) next(error);
+      if(!result) res.json(null);
+      const jwt=sign(req.body.email,SECRET);
+      res.cookie('jwt',jwt);
+    res.json(jwt);
```

Comment:
If you are putting the `jwt` in a cookie, why do you need to send it back in `res.json()`? Remember that, once set, cookies get sent back to server automatically with every request.

Also, sending `res.json(null)` is dangerous because your frontend will get errors if it tries to anything with the data.

Why not just set `jwt` cookie and send back helpful message to FE eg: `res.send('you are logged in successfully')`?